examples/actions/src/main.rs

```diff
@@ -0,0 +1,158 @@
+#![allow(rustdoc::missing_crate_level_docs)]
+
+use eframe::{App, Frame, egui};
+use egui::{Align2, Color32, Context, RichText};
+use egui_plot::{Line, Plot, PlotEvent, TooltipOptions};
+
+fn main() -> eframe::Result<()> {
+    eframe::run_native(
+        "Plot + actions + pins demo",
+        eframe::NativeOptions::default(),
+        Box::new(|_| Ok(Box::new(Demo::new()))),
+    )
+}
+
+struct Demo {
+    xs: Vec<f64>,
+    f1: Vec<f64>,
+    f2: Vec<f64>,
+    last_event: Option<String>,
+}
+
+impl Demo {
+    fn new() -> Self {
+        let n = 500;
+        let xs: Vec<f64> = (0..n).map(|i| i as f64 * 0.02).collect();
+        let f1: Vec<f64> = xs.iter().copied().map(f64::sin).collect();
+        let f2: Vec<f64> = xs
+            .iter()
+            .copied()
+            .map(|t| (t * 0.6 + 0.8).sin() * 0.8 + 0.2)
+            .collect();
+        Self {
+            xs,
+            f1,
+            f2,
+            last_event: None,
+        }
+    }
+}
+
+impl App for Demo {
+    fn update(&mut self, ctx: &Context, _frame: &mut Frame) {
+        egui::CentralPanel::default().show(ctx, |ui| {
+            ui.heading("Actions + Pins + Tooltip demo");
+            ui.add_space(6.0);
+
+            if let Some(s) = &self.last_event {
+                ui.label(RichText::new(s).size(16.0).monospace());
+            }
+
+
+            let (_resp, events) = Plot::new("demo_plot")
+                .allow_zoom(true)
+                .allow_scroll(true)
+                .allow_boxed_zoom(true)
+                .show_actions(ui, |plot_ui| {
+
+                 let pts1: Vec<[f64; 2]> = self
+                                                .xs
+    .iter()
+    .copied()
+    .zip(self.f1.iter().copied())
+    .map(<[f64; 2]>::from)
+    .collect();
+
+let pts2: Vec<[f64; 2]> = self
+    .xs
+    .iter()
+    .copied()
+    .zip(self.f2.iter().copied())
+    .map(<[f64; 2]>::from)
+    .collect();
+
+                    plot_ui.line(
+                        Line::new("f1", pts1)
+                            .color(Color32::from_rgb(200, 100, 100))
+                            .width(2.0),
+                    );
+
+                    plot_ui.line(
+                        Line::new("f2", pts2)
+                            .color(Color32::from_rgb(100, 160, 240))
+                            .width(2.0),
+                    );
+
+
+                    plot_ui.show_tooltip_with_options(&TooltipOptions::default());
+                });
+
+
+            for ev in events {
+                match ev {
+                    PlotEvent::BoundsChanged { old, new, cause } => {
+                        self.last_event = Some(format!(
```

Comment:
I think not all causes/events are implemented yet, I tried scrolling and cause showed as Programmatic